skipruntime-ts/ts/src/internals/skipruntime_process.ts

```diff
@@ -55,9 +55,12 @@ export async function runWithServer_(
           values: [[http, "", "read-write"]],
         });
       } else if (http.type == "post") {
-        update(http, {}).catch((e: unknown) => {
-          throw e;
-        });
+        const maybePromise = update(http, {});
+        if (typeof maybePromise === "object") {
+          maybePromise.catch((e: unknown) => {
+            throw e;
```

Comment:
It seems weird to create the `Promise` and never await it, should we turn the listener to an `async` function?
ESLint complains with `@typescript-eslint/no-misused-promises` but it seems that's what people do

Reply:
Yeah, I really don't know. This code is type-incorrect before and after this PR. The type of `ws.on` specifies that the `message` callback returns `void`, and here we return `void` on some paths and `Promise<void>` on others. But there is no way to wait on the `Promise`. Making the callback `async` allows it to be `await`ed, but then the callback returns a `Promise` which the `ws` library does not itself await. It seems to be just fundamentally wrong to call async code in the callbacks passed to `ws`, it is expecting sync code. Maybe there is an async websockets library we should use instead.